linalg/solvers.hpp

```diff
@@ -473,6 +473,38 @@ class SLBQPOptimizer : public OptimizationSolver
    virtual void Mult(const Vector &xt, Vector &x) const;
 };
 
+class BlockILU0 : public Solver
```

Comment:
This class badly needs some documentation. Something like the description in the current PR would be better than nothing.

Reply:
Definitely. I just added Doxygen comments that should help with this.